src/Kinepolis/KinepolisService.php

```diff
@@ -173,6 +182,19 @@ private function createNewMovie(ParsedMovie $parsedMovie): string
         $this->aggregateRepository->save($eventAggregate);
         $this->movieMappingRepository->create($eventId, $parsedMovie->getExternalId());
         $this->logger->info('Event created with id ' . $eventId);
+
+        // Movie pictures are normally never updated in the external API
+        // to avoid multiple needless updates, we only add an image during the creation
+        $uploadedImage = $this->client->getImage($token, $parsedMovie->getPoster());
+        $imageId = $this->imageUploader->upload(
+            $uploadedImage,
+            new MediaDescription($parsedMovie->getTitle()->toString()),
+            new CopyrightHolder('Kinepolis'),
+            new LegacyLanguage('nl')
+        );
+        $addImage = new AddImage($eventId, $imageId);
+        $this->commandBus->dispatch($addImage);
```

Comment:
It's a bit weird that the `createNewMovie` also does the dispatch of the `AddImage` command. Maybe that needs to become a separate method that can be called inside the `process` method. That new method can do the upload of the file and returns the `AddImage` command.